debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java

```diff
@@ -117,38 +136,97 @@ private PostgresReplicationConnection(Configuration config,
 
     protected void initPublication() {
         if (PostgresConnectorConfig.LogicalDecoder.PGOUTPUT.equals(plugin)) {
-            LOGGER.info("Initializing PgOutput logical decoder publication");
-            try {
-                String selectPublication = String.format("SELECT COUNT(1) FROM pg_publication WHERE pubname = '%s'", publicationName);
-                try (Statement stmt = pgConnection().createStatement(); ResultSet rs = stmt.executeQuery(selectPublication)) {
-                    if (rs.next()) {
-                        Long count = rs.getLong(1);
-                        if (count == 0L) {
-                            LOGGER.info("Creating new publication '{}' for plugin '{}'", publicationName, plugin);
-                            // Publication doesn't exist, create it.
-                            // todo: DBZ-766 - Change this to be restricted based on configured whitelist tables?
-                            // For situations where no publication exists, we likely cannot create it for all tables.
-                            // This is because postgres requires certain super user permissions to use "ALL TABLES".
-                            // We should restrict this to the configured tables here.
-                            stmt.execute(String.format("CREATE PUBLICATION %s FOR ALL TABLES;", publicationName));
-                        }
-                        else {
-                            LOGGER.trace(
-                                    "A logical publication named '{}' for plugin '{}' and database '{}' is already active on the server " +
-                                            "and will be used by the plugin",
-                                    publicationName, plugin, database());
+            if (PostgresConnectorConfig.AutoCreateMode.DISABLED.equals(publicationAutocreateMode)) {
```

Comment:
Shouldn't this be handled further below, in the swich block? It seems we'd trigger this exception in case the publication already exists?

Reply:
I went for a failing early approach, but I can see the argument for moving it into the switch, now that I look at it, that seems like a better place for it!